### DISCUSSION TITLE
Allow only US & Canada

### DISCUSSION THREAD

**Discussion by joshm1204** [kickstarter/rack-attack#522] [Questions (Q&A)]
Is there a method to allow only IP addresses originating from US & Canada?

**Comment by Nowaker** [comment]
I don't think Rack Attack is the best choice here. Instead, I'd suggest writing a generic middleware that performs a GeoIP check, and block anything that is not US/CA.

**Comment by dalezak** [comment]
@joshm1204 would something like this work?
```
Rack::Attack.blocklist 'blocklist geocoder location' do |req|
  Rack::Attack::Fail2Ban.filter req.ip, maxretry: 1, findtime: 30.seconds, bantime: 1.week do
    unless ['Canada', 'United States'].include?(Geocoder.search(req.ip).first.country)
      req.ip
    end
  end
end
```

**Comment by joshm1204** [comment]
I think so, but I ended up using cloudflare’s firewall, which works well.
I will be using rack:attack for other reasons though.

**Comment by Nowaker** [comment]
@dalezak I wouldn't recommend using this approach unless you explicitly configure Geocoder to use a [Local IP Address Lookup method](https://github.com/alexreisner/geocoder/blob/master/README_API_GUIDE.md#local-ip-address-lookups). Otherwise, every visiting IP except banned ones will be queried for geo-info every 30 seconds, which will have a negative effect on page loads (among other problems). Other than that, it's a very neat approach.
